src/aragonjs-wrapper.js

```diff
@@ -75,34 +75,40 @@ const pollEvery = (fn, delay) => {
 
 // Keep polling the main account.
 // See https://github.com/MetaMask/faq/blob/master/DEVELOPERS.md#ear-listening-for-selected-account-changes
-export const pollMainAccount = pollEvery((provider, onAccount) => {
-  const web3 = getWeb3(provider)
-  let lastAccount = null
-  let lastBalance = null
-  return {
-    request: () =>
-      getMainAccount(web3)
-        .then(account => {
-          if (!account) {
-            throw new Error('no account')
-          }
-          return web3.eth.getBalance(account).then(balance => ({
-            account,
-            balance: BigNumber(balance),
-          }))
-        })
-        .catch(() => {
-          return { account: null, balance: BigNumber(0) }
-        }),
-    onResult: ({ account, balance }) => {
-      if (account !== lastAccount || balance !== lastBalance) {
-        lastAccount = account
-        lastBalance = balance
-        onAccount(account, balance)
-      }
-    },
-  }
-}, POLL_DELAY_ACCOUNT)
+export const pollMainAccount = pollEvery(
+  (provider, { onAccount = () => {}, onBalance = () => {} } = {}) => {
+    const web3 = getWeb3(provider)
+    let lastAccount = null
+    let lastBalance = null
+    return {
+      request: () =>
+        getMainAccount(web3)
+          .then(account => {
+            if (!account) {
+              throw new Error('no account')
+            }
+            return web3.eth.getBalance(account).then(balance => ({
+              account,
+              balance: BigNumber(balance),
+            }))
+          })
+          .catch(() => {
+            return { account: null, balance: BigNumber(0) }
```

Comment:
Maybe balance could be set to `-1` here, so we can make the distinction between 0 and unknown?